plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -152,5 +156,38 @@ public static SymbolLookup get() {
         return j.getInjector().getInstance(SymbolLookup.class);
     }
 
+    /**
+     * Get the {@link Symbol} value(s) for the class of the given object, generally a {@link Descriptor}, if the annotation
+     * is present. If the object is in fact a {@link Describable}, we'll use its {@link Descriptor} class instead.
+     *
+     * @param o An object
+     * @return The {@link Symbol} annotation value(s) for the class (generally a {@link Descriptor} that object represents,
+     * or null if the annotation is not present.
+     */
+    public static Set<String> getSymbolValue(Object o) {
+        if (o instanceof Describable) {
+            return getSymbolValue(((Describable) o).getDescriptor().getClass());
+        } else {
+            return getSymbolValue(o.getClass());
+        }
+    }
+
+    /**
+     * Get the {@link Symbol} value(s) for the given class, if the annotation is present. Unlike {@link #getSymbolValue(Object)},
+     * this will not get the {@link Descriptor} for {@link Describable} classes.
+     *
+     * @param c A class.
+     * @return The {@link Symbol} annotation value(s) for the given class, or null if the annotation is not present.
+     */
+    public static Set<String> getSymbolValue(@Nonnull Class<?> c) {
+        Set<String> symbolValues = new HashSet<String>();
```

Comment:
:bug: will behave nondeterministically. Could use `TreeSet`, but probably better to use `LinkedHashSet` since there is a documented order to `value()`: first is preferred.